docs/user/exercises/programming-exercise-AI-proof-guidelines.inc

```diff
@@ -0,0 +1,491 @@
+
+   Example for an exercise where a UML diagram was given and the
+   attributes were not mentioned in the problem statement, which led to the AI struggling, as seen in the subsequent code comparison.
+
+   .. figure:: programming/AI-proof-exercises/JackOLantern.png
+      :alt: UML diagram for an exercise
+      :align: center
+      :width: 50%
+
+      UML diagram for an exercise showcasing the relationship between the classes as well as the content of the individual classes.
+
+   Generated solution:
+
+   .. code:: java
+
+      public class Pumpkin {
+
+
+          private double weight;
+          private String face;
+
+          public Pumpkin(double weight) {
+          this.weight = weight;
+          this.face = "";
+
+
+          }
+
+
+      }
+
+   Example solution:
+
+   .. code:: java
+
+      public class Pumpkin {
+          private double weight;
+          private String face;
+          private String type;
+          private boolean seeds;
+
+          public Pumpkin(String type,
+                  double weight) {
+              this.type = type;
+              this.weight = weight;
+              face = "";
+              seeds = true;
+          }
+
+      }
+
+
+-  **Depictions of functionalities:** Rather than explaining how a
+   concept works in theory, provide depictions of the functionality.
+   This encourages students to understand the basic code instead of
+   going to AI for answers. Examples that were used in previous
+   exercises:
+
+
+   .. figure:: programming/AI-proof-exercises/List-generics.png
+      :align: center
+      :alt: This figure showcases the functionality of a linked-list
+      :width: 50%
+
+
+      Depiction shows the functionality of a linked list with 3 integer values
+      as a basic concept for understanding linked lists. The task of this exercise
+      were to implement the common List functionalities from scratch rather than
+      rely on the standard methods from java. ListNode and LinkedList were
+      therefore own classes that needed to be filled with the required
+      functionalities.
+
+
+   This depiction provides an understanding of how a linked list should look
+   like after the add function was called.
+
+   .. figure:: programming/AI-proof-exercises/add.png
+      :align: center
+      :width: 50%
+
+
+      Figure shows what a list should look like after the add functionality was called. Picture from the Problem statement
+
+-  **Text in picture format:** Giving the text exercise instructions in
+   a picture format that can not be copied. Rather than writing long
+   inquiries into the AI, the students will be promoted to asking
+   reasonable questions and solve exercises on their own.
+
+   .. note::
+      Since there are already possibilities to copy text from pictures, the effect of
+      this measure are limited. More suitable for exam mode.
+
+-  **Java frameworks:** Using Java frameworks, such as JavaFX, is challenging for AI
```

Comment:
Out of curiosity, have you tested this? 
One of the earlier demo use cases of GPT3 (before ChatGPT was released) was to let it generate HTML UIs just from user input. So I'd be honestly surprised if ChatGPT has troubles with it, as long as you ask it correctly.
I'd assume with some decent prompting you might be able to get LLMs to generate decent UIs. Combine this with the advent of multi-modal models and eventually this should also be a trivial task.

Reply:
I tested it with the exercises from ITP from the last semester. From my experience it was fairly difficult for ChatGPT to reproduce the exact UI which was required. Sometimes there was need to put a vBox into another vBox as discussed and thought in the lecture. 
Feel free to give me more input on that so I can rewrite this part in a more suiting way